contents/node/nodejs.md

```diff
@@ -0,0 +1,202 @@
+
+### Benefits
+
+* Node.js is **fast** for I/O intensive tasks due to the non-blocking model.
+* As Node is **single-threaded**, we can avoid thread overheads and synchronization problems including deadlocks and race conditions.
+* Node has **good support** as npm (Node Package Manager) is the largest ecosystem of open source libraries in the world.
+* **Same language (JavaScript) for front and back-end development**. This makes it possible to share and reuse code between the front-end and back-end. Teams can be more efficient and less reliant on documentation as they can read the code directly. The team can also become more cross-functional as developers can contribute to both development.
+
+### Use cases
+
+Node.js is **good** for
+* **Processing high volumes of IO-bound requests**. A single instance of a Node server will be more efficient and can serve more requests with the same hardware than other usual servers (such as those written in Ruby on Rails). This makes the Node server faster and more scalable.
+* **Real time apps** where you have to process a large volume of requests with little delay. This includes instant messaging apps and collaborative editing apps where you can watch the document being modified live such as Trello and Google Docs. Node.js is a good choice as it can handle multiple client requests (even while waiting for responses).
+* **Single-page apps** where a lot of processing and rendering is done on the client's side and the backend server only needs to provide a simple API. Node can process many requests with low response times. In addition, you can reuse and share code, such as those for validation, between the client and server.
+
+However, Node.js is **not suitable** for
+* **CPU-intensive jobs**. Remember that Node.js is single-threaded. If the thread is busy doing CPU-heavy operations, it will not be able to process incoming requests timely. This removes the biggest advantage of Node. Perhaps, you should consider an alternative technology such as Go which supports concurrency.
+
+
+To find out more about when you should or should not use Node.js, checkout these articles by [netguru](https://www.netguru.co/blog/use-node-js-backend) and [Node.js foundation](https://medium.com/the-node-js-collection/why-the-hell-would-you-use-node-js-4b053b94ab8e).
+
+# Getting Started
+
+To install Node, simply download the installer from the [official Node.js website](https://nodejs.org/en/download/) based on your OS and run it. This should install both Node and npm. npm is a tool which will help you to search, install and manage node.js packages. The use of npm will be explained further in the [npm section](#node-package-manager-npm).
+
+# Writing a Hello World Node.js server
```

Comment:
May br remove this heading? The content can fit inside the `Getting Started` heading.